test/SolveByElim.lean

```diff
@@ -3,34 +3,54 @@ Copyright (c) 2021 Scott Morrison. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
 -/
+import Mathlib.Init.Data.Nat.Basic
+import Mathlib.Init.Logic
+import Std.Tactic.RCases
+import Mathlib.Tactic.Constructor
+import Mathlib.Tactic.PermuteGoals
 import Mathlib.Tactic.SolveByElim
 
 example (h : Nat) : Nat := by solve_by_elim
 example {α β : Type} (f : α → β) (a : α) : β := by solve_by_elim
 example {α β : Type} (f : α → α → β) (a : α) : β := by solve_by_elim
 example {α β γ : Type} (f : α → β) (g : β → γ) (a : α) : γ := by solve_by_elim
 example {α β γ : Type} (_f : α → β) (g : β → γ) (b : β) : γ := by solve_by_elim
-example {α : Nat → Type} (f : (n : Nat) → α n → α (n+1)) (a : α 0) : α 5 := by solve_by_elim
+example {α : Nat → Type} (f : (n : Nat) → α n → α (n+1)) (a : α 0) : α 4 := by solve_by_elim
```

Comment:
Why is this changed?

Reply:
This test is new, and would have failed in mathlib3 due to the even lower recursion bound there.